applications/FluidDynamicsApplication/custom_processes/two_fluids_inlet_process.cpp

```diff
@@ -0,0 +1,243 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Simon Wenczowski
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/openmp_utils.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "two_fluids_inlet_process.h"
+#include "fluid_dynamics_application_variables.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+TwoFluidsInletProcess::TwoFluidsInletProcess(
+    ModelPart& rModelPart,
+    Parameters& rParameters,
+    Process::Pointer p_dist_proc )
+    : Process(), mrInletModelPart(rModelPart) {
+
+    KRATOS_CHECK_VARIABLE_KEY( DISTANCE )
+    KRATOS_CHECK_VARIABLE_KEY( AUX_DISTANCE )
+    KRATOS_CHECK_VARIABLE_KEY( NODAL_AREA )
+
+    // checking all parameters that define the interface
+    Parameters default_parameters( R"(
+    {
+        "interface_normal"          : [0.0,1.0,0.0],
+        "point_on_interface"        : [0.0,0.25,0.0],
+        "inlet_transition_radius"   : 0.05
+    }  )" );
+
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+
+    // finding the complete model of the inlet model part
+    ModelPart& rRootModelPart = mrInletModelPart.GetRootModelPart();
+
+    // setting the parameters to the private data members of the class
+    mInterfaceNormal = rParameters["interface_normal"].GetVector();
+    mInterfacePoint = rParameters["point_on_interface"].GetVector();
+    mInletRadius = rParameters["inlet_transition_radius"].GetDouble();
+
+    // normalization of itnerface normal vector
+    if ( norm_2( mInterfaceNormal ) > 1.0e-7 ){
```

Comment:
Maybe 1e-7 is a too high tolerance value. Since this is user-definable, why don't you ask the user to provide the unit normal instead? (just asking)

Reply:
I am sorry but this is not entirely clear for me: Why is it too high? Or in other words: Why should any user find it completely necessary to define a normal vector as [1e-8, 0, 0]? On the other hand, I personally find it convenient to enter 45° as [1,1,0] instead of computing [sqrt(0.5), sqrt(0.5), 0]. The only reason of this threshold is too  avoid an accidental [0,0,0]. I am missing an important point?